app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/MapScreen.kt

```diff
@@ -116,7 +116,8 @@ fun createLocationReceiver(
         }
         val tempBeacon = findClosestBeacon(locationState.value!!, viewModel.beaconList.value)
         if (tempBeacon != null) {
-          viewModel.getRandomSong(tempBeacon.id)
+          viewModel.getRandomSongAndAddToProfile(tempBeacon.id)
+          viewModel.retrieveSongFromProfileAndAddToBeacon(tempBeacon.id)
           Log.d(
               "MapScreen",
               "retrieved song: ${song.value}") // TODO: the song is retrieved here if exist
```

Comment:
What is this TODO about ? Make sure to remove it if has been addressed or to create an issue for it please